src/test/java/teammates/test/cases/logic/EmailsTest.java

```diff
@@ -59,34 +64,45 @@ public static void classSetUp() throws Exception {
 
     @BeforeMethod
     public void caseSetUp() throws ServletException, IOException {
+        /* For Sendgrid testing */
+        fromEmail = "Admin@" + Config.inst().getAppId() + ".appspotmail.com";
+        fromName = "TEAMMATES Admin";
+        replyTo = "teammates@comp.nus.edu.sg";
         
-        InternetAddress internetAddress = new InternetAddress("Admin@"
-                + Config.inst().getAppId() + ".appspotmail.com",
-                "TEAMMATES Admin");
+        /* For MimeMessage testing */
+        InternetAddress internetAddress = new InternetAddress(fromEmail, fromName);
         from = internetAddress.toString();
-        replyTo = "teammates@comp.nus.edu.sg";
     }
 
     @Test
-    public void testGetEmailInfo() throws MessagingException {
-
-        Session session = Session.getDefaultInstance(new Properties(), null);
-        MimeMessage message = new MimeMessage(session);
-
+    public void testGetEmailInfo() throws MessagingException, JSONException {
         String email = "receiver@gmail.tmt";
         String from = "sender@gmail.tmt";
+        String subject = "email subject";
 
-        message.addRecipient(Message.RecipientType.TO, new InternetAddress(
-                email));
+        /* Test Sendgrid message */
+        Sendgrid message = new Sendgrid(Config.SENDGRID_USERNAME, Config.SENDGRID_PASSWORD);
 
-        message.setFrom(new InternetAddress(from));
-        String subject = "email subject";
+        message.addTo(email);
+        message.setFrom(from);
         message.setSubject(subject);
-        message.setContent("<h1>email body</h1>", "text/html");
+        message.setHtml("<h1>email body</h1>");
+
+        assertEquals("[Email sent]to=receiver@gmail.tmt|from=sender@gmail.tmt|subject=email subject",
+                                        Emails.getEmailInfo(message));
+
+        /* Test mime message */
```

Comment:
We have another way to divide test cases into sections right? For consistency, can we do the same here?